src/runtime/manifest-ast-nodes.ts

```diff
@@ -634,6 +635,12 @@ export interface Annotation extends BaseNode {
   simpleAnnotation?: string;
 }
 
+export interface ParameterizedAnnotation extends BaseNode{
+  kind: 'param-annotation';
+  simpleAnnotation: string;
+  parameter: string;
```

Comment:
It would be nice to know when this is a number or a number with units or a field name etc.

Either we can do some work now to make parameterized annotations more expression like or just leave a todo to explain their current use case and the limitations that come from it. (I'm trying to avoid being unable to reuse this work in future without having to rewrite the number with unit use here).

Reply:
Same question as before IIUC - should i rework this to be a specific TtlAnnotation node?

Reply:
I don't think so, Instead I would suggest parsing the parameter in the peg parser (rather than in js/ts code).

This would just change it from accepting a string to accepting a 'value with units' (or something like that). This will let us catch invalid syntax in the annotation at parse time.

Reply:
but not all annotations would necessarily be a numeric value with units - this works for ttl, but might working for something else, this is why i thought string would be more generic and preferable...

Reply:
I believe we should only add parsing capabilities when we know we will use them. I'm not blocking on this, but I would prefer to avoid allowing arbitrary strings for data that we later require to be valid.

In the past there was a similar issue where our parser accepted invalid json but the json parser would fail leading to user confusion. Again, this isn't blocking, when we know more about our needs for annotations we'll be able to write something more specific.